libbeat/docs/outputconfig.asciidoc

```diff
@@ -527,9 +565,56 @@ The password for connecting to Kafka.
 
 ===== topic
 
-The Kafka topic used for produced events. The setting can be a format string
+The Kafka topic used for produced events. The setting can be a *Format String*
 using any event field. To set the topic from document type use `%{[type]}`.
 
+===== topics
+
+Array of topic selector rules supporting conditionals, *Format String*
+based field access and name mappings. The first rule matching will be used to
+set the `topic` for the event to be published. If `topics` is missing or no
+rule matches, the `topic` field will be used.
+
+Rule settings:
+
+*`topic`*: The topic *Format String* to use. If the fields used are missing, the
+ rule fails.
+
+*`mapping`*: Dictionary mapping index names to new names
+
+*`default`*: Default string value if `mapping` does not find a match.
+
+*`when`*: Condition which must succeed in order to execute the current rule.
+
+===== key
+
+The Kafka event key. The event key must be unique and can be extracted from the
```

Comment:
keys in kafka are optional. if key is set and reused, the old content will be overwritten. By never changing content, keys can be used for some artificial kind of deduplication.